Added feature/brightness

vSebas opened the pull request.

vSebas committed:
Added feature/brightness

ivanacollg commented on src/test/data_augmentation_test.cpp:
```diff
@@ -22,6 +22,8 @@ int main( int argc, char** argv )
    //Apply Gaussian Blur filter
    data.GaussianBlur(kernel_size);
+   // Change brightness to 80
+   data.Brightness(80);
```
define el parametro como variable al inicio del programa

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -38,4 +38,24 @@ void DataAugmentation::GaussianBlur(const int &kernel){
+  // https://docs.opencv.org/3.4/d3/dc1/tutorial_basic_linear_transform.html
+  int i,j;
+  out_ = cv::Mat::zeros( in_.size(), in_.type() );
+  double alpha_ = 1.0;
```
varibles de funciones no llevan guion bajo: alpha

ivanacollg replied:
Comnenta tambien lo que controla alpha: contraste y agregalo como parametro a la funcion, igual que brightness

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -38,4 +38,24 @@ void DataAugmentation::GaussianBlur(const int &kernel){
+  int i,j;
+  out_ = cv::Mat::zeros( in_.size(), in_.type() );
+  double alpha_ = 1.0;
+  int beta_ = brightness;
```
sin guion bajo beta. No veo la necesidad de declarar otra variable beta, si podemos usar el argumento brightness directamente

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -38,4 +38,24 @@ void DataAugmentation::GaussianBlur(const int &kernel){
+void DataAugmentation::Brightness(const int brightness){
+  // Full explanation can be found in: 
+  // https://docs.opencv.org/3.4/d3/dc1/tutorial_basic_linear_transform.html
+  int i,j;
```
Estas declarando las variables i y j dos veces. Sugiero decrlarar aqui i y c, para no declararlas adentro de los for

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -38,4 +38,24 @@ void DataAugmentation::GaussianBlur(const int &kernel){
+  }
+  cv::namedWindow("Brightness_value: "+std::to_string(beta_),CV_WINDOW_NORMAL); 
+  cv::imshow("Brightness_value: "+std::to_string(beta_), out_);
+  cv::imwrite("../../imgs/Brightness:"+ std::to_string(beta_)+".png", out_);
```
paths de outputs sin mayusculas

vSebas replied:
Done

ivanacollg reviewed (changes requested):
Agrega tu nombre como coautor al header y la fecha que se modifico por ultima vez

vSebas committed:
Made these changes:
Alpha and beta were passed as class method parameters
Output path now is in lower case
Changed the name of the class method

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -40,22 +40,20 @@ void DataAugmentation::GaussianBlur(const int &kernel){
 }
 
-void DataAugmentation::Brightness(const int brightness){
+void DataAugmentation::ContrastBrightness(const double alpha, const int beta){
```
En lugar de llamar los parametros aloha o beta, llamalos dependiendo  lo que controlan. No estoy segura cual es cual, pero uno controla brightness y el otro contraste.

vSebas replied:
Done

ivanacollg commented on src/test/data_augmentation_test.cpp:
```diff
@@ -16,14 +16,16 @@ int main( int argc, char** argv )
    cv::Mat out;
    int kernel_size=5;
+   double alpha=1.0;
+   int beta=80;
```
Agrega espacios antes y despues de signo =

vSebas replied:
Done

ivanacollg commented on src/test/data_augmentation_test.cpp:
```diff
@@ -16,14 +16,16 @@ int main( int argc, char** argv )
    cv::Mat in; 
    cv::Mat out;
    int kernel_size=5;
```
Yo se que yo hice esto, pero si puedes agregar espacios antes y despues de los signos = en general en todo.

vSebas replied:
Mejor lo cambio en master

ivanacollg commented on src/data_augmentation.h:
```diff
@@ -35,8 +35,9 @@ class DataAugmentation {
   //
-  // @param value[in]: how bright the image will be (-100,100)
-  void Brightness(const int brightness);
+  // @param alpha[in]: contrast control
```
Si puedes agregar al comentario el rango acceptable, ejemplo: (0-3)

vSebas replied:
Done

ivanacollg commented on src/test/data_augmentation_test.cpp:
```diff
@@ -16,14 +16,16 @@ int main( int argc, char** argv )
    cv::Mat in; 
    cv::Mat out;
    int kernel_size=5;
+   double alpha=1.0;
```
Agrega comentario antes de los dos  //ContrastBrightness params

vSebas replied:
Done

ivanacollg commented on src/test/data_augmentation_test.cpp:
```diff
@@ -16,14 +16,16 @@ int main( int argc, char** argv )
    cv::Mat in; 
    cv::Mat out;
    int kernel_size=5;
```
Agrega comentario // Bluss Params arriba de int kernel_size

vSebas replied:
Mejor lo cambio en master

vSebas replied:
Done

vSebas committed:
Changes:
Added comments
Changed the name of params
Improved syntax

vSebas committed:
Changes:
Changed the name of params

vSebas committed:
Updatted brief

vSebas committed:
Added Ivana's mail
